hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/payload/SqlTypedRecord.scala

```diff
@@ -53,6 +53,11 @@ object SqlTypedRecord {
 
   private val avroDeserializerCache = CacheBuilder.newBuilder().build[Schema, HoodieAvroDeserializer]()
 
+  private val avroDeserializerCacheLocal = new ThreadLocal[Cache[Schema, HoodieAvroDeserializer]] {
+    override def initialValue(): Cache[Schema, HoodieAvroDeserializer] =
+      CacheBuilder.newBuilder().maximumSize(16).build[Schema, HoodieAvroDeserializer]()
```

Comment:
this looks not used at all? `avroDeserializerCache` still used for storing the deserializer

Reply:
> this looks not used at all? `avroDeserializerCache` still used for storing the deserializer
> So, what are we try to fix here ? The schema key in the cache does not work in multi-thread use case ?

reuse the same schema avroDeserializer in diff thread will cause the result record diff from input record.
for example:
schema(key is id) : id int, name string, age int
there has two thread task use  SqlTypedRecord to get sqlRow int the same time.
task one record:   1,  'one',  18
task two record:   2,  'two',  19
if reuse the sanme avroDeserializer, after deserialize in the same time, may has the following results:
task one result:    1,  'two', 19
task two result:    2,  'two', 19

Reply:
or Cache key use  `schema + thread_id` I think can solve this question

Reply:
@KnightChess what i meant was, the current implementation does not use the cache object it created from `initialValue()`. The objects are put into `avroDeserializerCache`, which is not from the ThreadLocal

Reply:
@xushiyan Oh, I get. the `avroDeserializerCache` is unnecessary. here, it only update global, not be used in single thread